src/interpreter.rs

```diff
@@ -1,10 +1,63 @@
-use crate::ast::{BinaryOperator, Expression, LogicalExpression, Predicate, Value};
+use crate::ast::{
+    BinaryOperator, Expression, LogicalExpression, Predicate, Route, RouteLogicalOperators,
+    RouteTerm, Value,
+};
 use crate::context::{Context, Match};
 
 pub trait Execute {
     fn execute(&self, ctx: &mut Context, m: &mut Match) -> bool;
 }
 
+pub trait Convert {
+    fn convert(&self, route: &mut Route);
+}
+
+impl Convert for Expression {
+    fn convert(&self, route: &mut Route)  {
+        match self {
+            Expression::Logical(logic_exp) => match logic_exp.as_ref() {
+                LogicalExpression::And(l, r) => {
+                    l.convert(route);
+                    r.convert(route);
+                    route
+                        .stack
+                        .push(RouteTerm::LogicalOperator(RouteLogicalOperators::And));
+                }
+                LogicalExpression::Or(l, r) => {
+                    l.convert(route);
+                    r.convert(route);
+                    route
+                        .stack
+                        .push(RouteTerm::LogicalOperator(RouteLogicalOperators::Or));
+                }
+                LogicalExpression::Not(r) => {
+                    r.convert(route);
+                    route
+                        .stack
+                        .push(RouteTerm::LogicalOperator(RouteLogicalOperators::Not));
+                }
+            },
+            Expression::Predicate(p) => {
+                let mut predicate = Predicate {
+                    lhs: crate::ast::Lhs {
+                        var_name: p.lhs.var_name.clone(),
+                        transformations: Vec::new(),
```

Comment:
Could we just init this like `transformations: p.lhs.transformations.clone(),`, so that we don't need the iteration of transformation assignment below.

Reply:
not really.